### ISSUE TITLE
Generate reference documentation for modules and steps

### ISSUE THREAD

**Issue by sparshev** [griddynamics/mpl#88]
Probably will be a good addition to the library a way to automatically generate/validate documentation (and module headers) out of used `CFG` / `OUT` variables. Need to check a way to do that automatically by using some existing (groovydoc?) framework.

Reporter: @Napo2k

**Comment by Napo2k** [comment]
I am attempting this (albeit on-the-side) with `groovydoc` and so far I have not been able to generate proper documentation for a module (which is my main interest for this). I think the main problem is that the resource module files are not really classes, but scripts, which makes the generated documentation a bit awkward (i.e., it _does_ generate the HTMLs for them but mentions constructors and destructors, for instance, which is pointless in that case)
What I want is to tell the user "this is how the config looks for this module", as well as adding some examples, perhaps.
But I will invest more time soon (TM) on this and get back here with my findings

**Comment by sparshev** [comment]
Yeah, usually I prepend my modules with comment about the inputs and outputs, but defensively it will be better to generate such doc and validate that the required comment is here and the format is ok.
